### PR TITLE
[easy] Add `RecursionChallenge` type for `ProverProof`

### PR DESCRIPTION
**Pull request by querolita**
This PR adds a type called `RecursionChallenge` to substitute the tuple of `prev_evals` inside `ProverProof`. This was proposed in the older [PR for serialization](https://github.com/o1-labs/proof-systems/pull/570) that was substituted by [this one](https://github.com/o1-labs/proof-systems/pull/599/files), already merged. If there's interest for this change, I already had [this PR](https://github.com/MinaProtocol/mina/pull/11130) addressing the mina side, that I would reopen.

It also compresses the two test files about serialization into one file only, as they were redundant.

### CODE REVIEW

File: kimchi/src/prover.rs [hunk 1]
```diff
@@ -1050,7 +1051,8 @@ where
 
         let polys = prev_challenges
             .iter()
-            .map(|(chals, comm)| {
+            .map(|c| {
+                let RecursionChallenge { chals, comm } = c;
```
Comment (mrmr1993) [inline kimchi/src/prover.rs:1055]: Nit: inline this pattern in the argument?

File: kimchi/src/proof.rs [hunk 2]
```diff
@@ -35,6 +36,9 @@ pub struct LookupEvaluations<Field> {
 }
 
 // TODO: this should really be vectors here, perhaps create another type for chunked evaluations?
+/// Polynomial evaluations contained in a `ProverProof`.
```
Comment (mrmr1993) [inline kimchi/src/proof.rs:39]: Please add unrelated doc-comments in their own PR!

File: kimchi/src/verifier.rs [hunk 3]
```diff
@@ -72,9 +72,10 @@ where
     ) -> Vec<Vec<Vec<ScalarField<G>>>> {
         self.prev_challenges
             .iter()
-            .map(|(chals, _poly)| {
+            .map(|c| {
                 // No need to check the correctness of poly explicitly. Its correctness is assured by the
                 // checking of the inner product argument.
+                let RecursionChallenge { chals, comm: _ } = c;
```
Comment (mrmr1993) [inline kimchi/src/verifier.rs:78]: Nit: inline in function argument?

Comment (querolita) [reply kimchi/src/verifier.rs]: Thanks!

File: kimchi/src/prover.rs (hunk 1 above)
Comment (querolita) [reply kimchi/src/prover.rs]: Thanks!

File: kimchi/src/proof.rs (hunk 2 above)
Comment (querolita) [reply kimchi/src/proof.rs:39]: 🤦🏻 oh, I usually forget

**Pull request merged**